pkg/server/download.go

```diff
@@ -289,12 +285,24 @@ func (s *server) downloadFileFromPresignedUrl(ctx context.Context, url string, t
 	if _, err = io.Copy(f, progressReader); err != nil {
 		log.Infof("Error while downloading: %v", err)
 		fmt.Println(" - Download cancelled")
-		_ = os.Remove(tempPath)
-		return 0, err
-	}
 
-	err = os.Rename(tempPath, targetLocation)
-	if err != nil {
+		// Repopulate dummy file with the packageID
+		f, openErr := os.OpenFile(targetLocation, os.O_TRUNC|os.O_WRONLY, 0644)
+		if openErr != nil {
+			log.Warnf("Failed to write PackageID: %v", openErr)
+			return 0, err
+		}
+		defer func(f *os.File) {
+			err := f.Close()
+			if err != nil {
+
+			}
```

Comment:
was something supposed to go in this conditional? if not you can probably remove it and change the line above to `_ = f.Close()` or just `f.Close()`